### PR TITLE
Issue/46 get plugin improved

### LINKED ISSUE #46: Simplify get_plugin to make use of improved module loading
get_plugin could now make use of the normal import system

### LINKED ISSUE #49: get_plugin and import inmanta_plugins produce inconsistent results
# Problem

The code loading by `get_plugin`  and `import inmanta_plugins` is considered different.

related to #46 

# Example

```python
class TestException(Exception):
   pass

@plugin
get_exception():
   return TestException
```

```python
def test_plugin(project):
   from  inmanta_plugins.unittest get TestException
   testexception = project.get_plugin("get_exception)()
   assert TestException == testexception
```

This will fail with 

```
AssertionError: assert <class 'inmanta_plugins.unittest.__init__.TestException'> == <class 'inmanta_plugins.unittest.TestException'>
```

observe that the plugin loads the code in the wrong package: it has the filename in it, even if it is `__init__`

# Notes

There could be more subtle problems when doing a compile, as it may re-load the code:

What could be tested

```python
def test_plugin(project):
   from  inmanta_plugins.unittest import get TestException
   testexception = project.get_plugin("get_exception)()
   assert TestException is testexception
   
   project.compile("import unittest")

   from  inmanta_plugins.unittest import get TestException
   assert TestException is testexception
   testexception2 = project.get_plugin("get_exception)()
   assert TestException is testexception2
   assert testexception is testexception2

```

### PR DESCRIPTION
**Pull request by sanderr**
# Description

- Improves `Project._load_plugins()` implementation to make sure it is compatible with the compiler's import mechanism
- Adds `inmanta_plugins` fixture to make abstraction of required module reloading when the compiler project is reset

closes #46 
closes #49

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [x] Attached issue to pull request
- [x] Changelog entry
- [x] Code is clear and sufficiently documented
- [x] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Correct, in line with design
- [x] End user documentation is included or an issue is created for end-user documentation

# Reviewer Checklist:

- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Code is clear and sufficiently documented
- [ ] Correct, in line with design

### CODE REVIEW

**Comment by sanderr** [comment]
@arnaudsjs I don't know how familiar you are with this part of the code. I mostly added you as a reviewer to review the documentation and the usability of the fixture I added. Of course, feel free to review the rest as well.

File: pytest_inmanta/plugin.py [hunk 1]
```diff
@@ -247,14 +255,56 @@ def __init__(self, uri):
...
+            raise AttributeError("No inmanta module named %s" % name)
+        return submodules[fq_mod_name]
+
+
+class InmantaPluginsImporter:
+    def __init__(self, project: "Project") -> None:
+        self.project: Project = project
+        self.loader: InmantaPluginsImportLoader = InmantaPluginsImportLoader(self)
+        # expose module imports from self.modules.some_module
+        self.modules: InmantaPluginsImportLoader = self.loader
```
Comment (arnaudsjs) [inline pytest_inmanta/plugin.py:279]: Are these variables used anywhere?

Comment (sanderr) [reply pytest_inmanta/plugin.py]: Indeed they aren't. `self.loader` should be though. The other I forgot to remove. Fixed both.

File: pytest_inmanta/plugin.py [hunk 2]
```diff
@@ -247,14 +254,54 @@ def __init__(self, uri):
...
+    def get_submodules(self, module_name: str) -> Optional[Dict[str, ModuleType]]:
+        inmanta_project: module.Project = module.Project.get()
+        if not inmanta_project.loaded:
+            raise Exception(
+                "Dynamically importing from inmanta_plugins requires a loaded inmanta.module.Project. Make sure to use the"
+                " project fixture."
+            )
+        modules: Dict[str, module.Module] = inmanta_project.get_modules()
+        importlib.invalidate_caches()
+        loader.configure_module_finder(self.project.module_path)
```
Comment (arnaudsjs) [inline pytest_inmanta/plugin.py:287]: I don't think this line is required. When `project.load()` is invoked, the module_finder should be configured correctly. Is that correct?

Comment (sanderr) [reply pytest_inmanta/plugin.py:287]: I agree, will fix.

**Pull request merged**